### PR TITLE
Backport #115197

### PR DESCRIPTION
**Pull request by pxsalehi**
Backport https://github.com/elastic/elasticsearch/pull/115197

Closes https://github.com/elastic/elasticsearch/issues/115318

### CODE REVIEW

**Comment by pxsalehi** [comment]
due to the language level used in 7.x, it seems we can't use that version check. Since it's 7.x, probably ok to just always flush.

**Review by ywangd — APPROVED** [review]
LGTM pending green CI

**Comment by pxsalehi** [comment]
@elasticmachine update branch

**Review by idegtiarenko — APPROVED** [review]

File: test/framework/src/main/java/org/elasticsearch/repositories/blobstore/AbstractBlobContainerRetriesTestCase.java [hunk 1]
```diff
@@ -403,9 +403,7 @@ protected void sendIncompleteContent(HttpExchange exchange, byte[] bytes) throws
         if (bytesToSend > 0) {
             exchange.getResponseBody().write(bytes, rangeStart, bytesToSend);
         }
-        if (randomBoolean()) {
-            exchange.getResponseBody().flush();
-        }
+        exchange.getResponseBody().flush();
```
Comment (idegtiarenko) [inline test/framework/src/main/java/org/elasticsearch/repositories/blobstore/AbstractBlobContainerRetriesTestCase.java:406]: Probably worth a comment on why it is needed

**Comment by pxsalehi** [comment]
@elasticmachine update branch

**Pull request merged**